editor/editor_node.cpp

```diff
@@ -1042,34 +1026,54 @@ void EditorNode::_resources_reimported(const Vector<String> &p_resources) {
 }
 
 void EditorNode::_sources_changed(bool p_exist) {
-	if (waiting_for_first_scan) {
-		waiting_for_first_scan = false;
+	if (!waiting_for_first_scan) {
+		return;
+	}
```

Comment:
Here, I just early exited instead of wrapping the whole contents of the function inside an if block.